libraries/AP_IOMCU/fw_uploader.cpp

```diff
@@ -281,7 +307,11 @@ bool AP_IOMCU::erase()
     debug("erase...");
     send(PROTO_CHIP_ERASE);
     send(PROTO_EOC);
-    return get_sync(10000);
+#if HAL_IOMCU_CHIBIOS_BOOTLOADER
+    return get_sync(20000); // 20s timeout for erase
+#else
+    return get_sync(10000); // 10s timeout for erase
+#endif
```

Comment:
Does the erase take longer just because of `HAL_IOMCU_CHIBIOS_BOOTLOADER`?

Should this instead be `HAL_IOMCU_FW_FLASH_SIZE > (0x10000 - 0x1000)`?

Reply:
the reason is flash size, so I guess it can be changes to be based on that.

Reply:
fixed